src/Sulu/Component/Webspace/Manager/WebspaceManager.php

```diff
@@ -124,6 +125,35 @@ public function findUrlsByResourceLocator(
     /**
      * {@inheritdoc}
      */
+    public function findUrlByResourceLocator(
+        $resourceLocator,
+        $environment,
+        $languageCode,
+        $webspaceKey = null,
+        $domain = null,
+        $scheme = 'http'
+    ) {
+        $urls = [];
+        $portals = $this->getWebspaceCollection()->getPortalInformations($environment);
+        foreach ($portals as $url => $portalInformation) {
+            $sameLocalization = $portalInformation->getLocalization()->getLocalization() === $languageCode;
+            $sameWebspace = $webspaceKey === null || $portalInformation->getWebspace()->getKey() === $webspaceKey;
+            $url = rtrim(sprintf('%s://%s%s', $scheme, $url, $resourceLocator), '/');
+            if ($sameLocalization && $sameWebspace && $this->isFromDomain($url, $domain)) {
+                if ($portalInformation->isMain()) {
+                    array_unshift($urls, $url);
+                } else {
+                    $urls[] = $url;
+                }
```

Comment:
what do you think about this logic? @chirimoya 

It returns a main url it there is one for this domain and locale otherwise it returns the first url which is found from this domain.